lib/spack/spack/bootstrap.py

```diff
@@ -20,17 +27,33 @@
 from spack.util.environment import EnvironmentModifications
 
 
+def spec_for_current_python():
+    """For bootstrapping purposes we are just interested in the Python
+    minor version (all patches are ABI compatible with the same minor)
+    and on whether ucs4 support has been enabled for Python 2.7
+
+    See:
+      https://www.python.org/dev/peps/pep-0513/
+      https://stackoverflow.com/a/35801395/771663
+    """
+    version_str = '.'.join(str(x) for x in sys.version_info[:2])
```

Comment:
There's a line in this file later in `make_module_available()`: `spec.constrain('^python@%d.%d' % sys.version_info[:2])`. Where is spack eventually using this spec?

Reply:
> Where is spack eventually using the spec we generate in this function?

In the `spack_python_interpreter` context manager at line 56. That context manager is used with others in the `ensure_bootstrap_configuration` context manager used in `asp.py`. That said it might be a good idea to use the spec returned here also in:
```
spec.constrain('^python@%d.%d' % sys.version_info[:2])
```
I'll check on that, thanks!

Reply:
done in b0a61fb67f21d02bd31c01f3bf22216ee58e7df5